### ISSUE TITLE
Training AlexNet structure on entire ImageNet in Tensorflow 2.0, But why the loss and accuracy are all flat?

### ISSUE THREAD

**Issue by anejad** [tensorflow/tensorflow#43503]
I am trying to replicate AlexNet model on ImageNet (for learning purposes). The dataset is 1.2 million ImageNet dataset with 50K validation. 
Model is a sequential keras CNN model, and I am training this model on 8 GPUs. 

The code can be seen here:

[Code on GitHub](https://github.com/anejad/Convolutional-Neural-Network-Champions/blob/master/AlexNet/AlexNet_Tensorflow_Full.py).

However, when I start to train this model (8 GPUs, Batch size=32*nGPU) I get flat loss and accuracy:
Epoch 14/90
5004/5004 [==============================] - 748s 150ms/step - loss: 6.9080 - accuracy: 8.8055e-04 - val_loss: 6.9078 - val_accuracy: 9.6154e-04
Epoch 15/90
5004/5004 [==============================] - 747s 149ms/step - loss: 6.9080 - accuracy: 9.6798e-04 - val_loss: 6.9078 - val_accuracy: 3.2051e-04
Epoch 16/90
5004/5004 [==============================] - 745s 149ms/step - loss: 6.9080 - accuracy: 0.0010 - val_loss: 6.9077 - val_accuracy: 0.0013
Epoch 17/90
5004/5004 [==============================] - 748s 149ms/step - loss: 6.9080 - accuracy: 9.5559e-04 - val_loss: 6.9078 - val_accuracy: 8.0128e-04
Epoch 18/90
5004/5004 [==============================] - 749s 150ms/step - loss: 6.9080 - accuracy: 8.8055e-04 - val_loss: 6.9078 - val_accuracy: 0.0014
Epoch 19/90
5004/5004 [==============================] - 750s 150ms/step - loss: 6.9080 - accuracy: 9.4924e-04 - val_loss: 6.9078 - val_accuracy: 0.0013
Epoch 20/90
5004/5004 [==============================] - 750s 150ms/step - loss: 6.9080 - accuracy: 9.3051e-04 - val_loss: 6.9078 - val_accuracy: 0.0016
Epoch 21/90
5004/5004 [==============================] - 748s 149ms/step - loss: 6.9080 - accuracy: 0.0010 - val_loss: 6.9078 - val_accuracy: 0.0011
Epoch 22/90
5004/5004 [==============================] - 749s 150ms/step - loss: 6.9080 - accuracy: 9.8047e-04 - val_loss: 6.9078 - val_accuracy: 0.0014
Epoch 23/90
5004/5004 [==============================] - 748s 149ms/step - loss: 6.9080 - accuracy: 7.3067e-04 - val_loss: 6.9078 - val_accuracy: 9.6154e-04
Epoch 24/90
5004/5004 [==============================] - 750s 150ms/step - loss: 6.9080 - accuracy: 0.0011 - val_loss: 6.9078 - val_accuracy: 0.0014
Epoch 25/90
5004/5004 [==============================] - 750s 150ms/step - loss: 6.9080 - accuracy: 0.0010 - val_loss: 6.9078 - val_accuracy: 1.6026e-04

Any ideas how to fix this or what causes the learning to be flat???

**Comment by gowthamkpr** [comment]
This is very similar to this issue [here](https://stackoverflow.com/questions/63236540/tensorflow-loss-and-accuracy-stay-flat-training-cnn-on-image-classification). Please take a look at it.

**Comment by anejad** [comment]
> This is very similar to this issue [here](https://stackoverflow.com/questions/63236540/tensorflow-loss-and-accuracy-stay-flat-training-cnn-on-image-classification). Please take a look at it.

This case is completely different than mine. I am trying to train the Tensorflow model on 1000 classes so I cannot use Binary mode in my ImageDataGenerator. Am I missing anything?

**Comment by anejad** [comment]
This is the code:
```
def Pipelines(train_directory,validation_directory,BATCH_SIZE,SEED,image_height, image_width,classes):
    train_datagen = ImageDataGenerator(
                  rescale=1./255,) 
                  #rotation_range=10,
                  #width_shift_range=0.1,
                  #height_shift_range=0.1,
                  #shear_range=0.1,
                  #zoom_range=0.1)

    train_generator = train_datagen.flow_from_directory(train_directory,\
                                                    target_size=(image_height, image_width),\
                                                    color_mode="rgb",\
                                                    batch_size=BATCH_SIZE,\
                                                    seed=SEED,\
                                                    shuffle=True,\
                                                    classes=classes,
                                                    class_mode="categorical")
    valid_datagen=ImageDataGenerator(rescale=1./255)

    valid_generator = valid_datagen.flow_from_directory(validation_directory,
                                                            target_size=(image_height, image_width),
                                                            color_mode="rgb",
                                                            batch_size=BATCH_SIZE,
                                                            seed=SEED,
                                                            shuffle=True,
                                                            classes=classes,
                                                            class_mode="categorical",
                                                            )
return train_generator,valid_generator

def AlexNet( input_shape, num_classes):
    initializer = tf.keras.initializers.GlorotNormal()
    model = Sequential(name='AlexNet')
    model.add(Conv2D(96, kernel_size=(11,11), strides= 4,
                    padding= 'valid', activation= 'relu',
                    input_shape= input_shape, kernel_initializer= initializer))
    model.add(MaxPooling2D(pool_size=(3,3), strides= (2,2),
                          padding= 'valid', data_format= None))

    model.add(Conv2D(256, kernel_size=(5,5), strides= 1,
                    padding= 'same', activation= 'relu',
                    kernel_initializer= initializer))
    model.add(MaxPooling2D(pool_size=(3,3), strides= (2,2),
                          padding= 'valid', data_format= None)) 

    model.add(Conv2D(384, kernel_size=(3,3), strides= 1,
                    padding= 'same', activation= 'relu',
                    kernel_initializer= initializer))

    model.add(Conv2D(384, kernel_size=(3,3), strides= 1,
                    padding= 'same', activation= 'relu',
                    kernel_initializer= initializer))

    model.add(Conv2D(256, kernel_size=(3,3), strides= 1,
                    padding= 'same', activation= 'relu',
                    kernel_initializer= initializer))

    model.add(MaxPooling2D(pool_size=(3,3), strides= (2,2),
                          padding= 'valid', data_format= None))

    model.add(Flatten())
    model.add(Dense(4096, activation= 'relu'))
    model.add(Dropout(rate=0.5))
    model.add(Dense(4096, activation= 'relu'))
    model.add(Dropout(rate=0.5))
    model.add(Dense(1000, activation= 'relu'))
    model.add(Dense(num_classes, activation= 'softmax'))

    model.compile(optimizer= tfa.optimizers.SGDW(learning_rate=0.01,momentum=0.9,weight_decay=0.0005,nesterov=False),
                loss='categorical_crossentropy',
                metrics=['categorical_accuracy'])
    return model

def TrainModel(model,valid_generator,train_generator,EPOCHS,BATCH_SIZE,PATIENCE_FRACTION):   
    early_stopping=EarlyStopping(monitor='val_loss',min_delta=0.0005,patience=int(EPOCHS*PATIENCE_FRACTION))
    checkpoint=ModelCheckpoint(filepath='./models/checkpoint_saves/AlexNet_checkpoint_save.h5',\
                               monitor='val_loss',save_weights_only=False)
    callback_list = [early_stopping,checkpoint]
    history=model.fit(train_generator,
                    epochs=EPOCHS,
                    validation_data=valid_generator,
                    callbacks=callback_list,
                    steps_per_epoch=train_generator.samples//BATCH_SIZE,
                    validation_steps= valid_generator.samples//BATCH_SIZE,
                    verbose=1)
    # laoding the best model 
    checkpoint_model=tf.keras.models.load_model('./models/checkpoint_saves/AlexNet_checkpoint_save.h5')
    return checkpoint_model,history

# Train model
bs=BATCH_SIZE
if strategy.num_replicas_in_sync>0:
    bs=BATCH_SIZE*strategy.num_replicas_in_sync
model,history=TrainModel(model,valid_generator,train_generator,EPOCHS,bs,PATIENCE_FRACTION)
```

**Comment by nikitamaia** [comment]
Hi @anejad, are you still facing this problem? Also what strategy are you using?
